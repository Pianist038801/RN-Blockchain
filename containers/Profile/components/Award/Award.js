import React from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import VerifiedIcon from '../VerifiedIcon'
import NameBlock from '../NameBlock'
import DateBlock from '../DateBlock'
import styles from '../../../../style.js'

const Award = (props) => {
  const {
    name,
    line2,
    date,
    award
  } = props

  return (
    <View style={styles.profileEntityContainer}>
      <VerifiedIcon type="award" entity={award} />
      <NameBlock name={name} line2={line2} />
      <DateBlock date={date} />
    </View>
  )
}

Award.propTypes = {
  name: PropTypes.string.isRequired,
  line2: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]),
  award: PropTypes.shape({
    pending_verification: PropTypes.bool,
    verified: PropTypes.bool
  }),
  date: PropTypes.string.isRequired
}

export default Award
