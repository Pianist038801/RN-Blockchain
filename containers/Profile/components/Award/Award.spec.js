import React from 'react'
import renderer from 'react-test-renderer'
import Award from './index'

describe('containers/Profile/components/Award', () => {
  it('renders correctly', () => {
    const props = {
      award: {
        verified: true
      },
      name: 'hello',
      line2: 'Lorem ipsum',
      date: '2017-11-04'
    }

    const tree = renderer.create(
      <Award {...props} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
