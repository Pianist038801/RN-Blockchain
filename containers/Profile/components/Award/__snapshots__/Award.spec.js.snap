// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/Award renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#fff",
      "borderWidth": 1,
      "flex": 0.5,
      "flexDirection": "row",
      "marginBottom": 5,
      "padding": 22.5,
      "width": 600,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 75,
        "resizeMode": "contain",
        "width": 75,
      }
    }
  />
  <View
    style={
      Object {
        "marginLeft": 22.5,
        "width": 360,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#569cee",
          "fontWeight": "bold",
        }
      }
    >
      hello
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Lorem ipsum
    </Text>
  </View>
  <View
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
      2017
    </Text>
  </View>
</View>
`;
